packages/core/src/blocks/Table/block.ts

```diff
@@ -394,6 +395,87 @@ export const createTableBlockSpec = () =>
+      // and all cells are selected. Uses a separate extension as it needs
+      // priority over keyboard handlers in the `TableExtension`'s
+      // `tableEditing` plugin.
+      createBlockNoteExtension({
+        key: "table-keyboard-delete",
+        keyboardShortcuts: {
+          Backspace: ({ editor }) => {
+            if (!(editor.prosemirrorState.selection instanceof CellSelection)) {
+              return false;
+            }
+
+            const block = editor.getTextCursorPosition().block;
+            const content = block.content as TableContent<any, any>;
+
+            const rows = content.rows.length;
+            let cols = 0;
+
+            for (let rowIndex = 0; rowIndex < content.rows.length; rowIndex++) {
+              for (
+                let cellIndex = 0;
+                cellIndex < content.rows[rowIndex].cells.length;
+                cellIndex++
+              ) {
+                const cell = content.rows[rowIndex].cells[cellIndex];
+
+                // Counts number of columns in table from first row.
+                if (rowIndex === 0) {
+                  const colSpan =
+                    "type" in cell && cell.props.colspan
+                      ? cell.props.colspan
+                      : 1;
+                  cols += colSpan;
+                }
+
+                // Returns `false` if any cell isn't empty.
+                if (
+                  ("type" in cell && cell.content.length > 0) ||
+                  (!("type" in cell) && cell.length > 0)
+                ) {
+                  return false;
+                }
+              }
+            }
+
+            // Need to use ProseMirror API to check if selection spans table.
+            const anchorCellColIndex =
+              editor.prosemirrorState.selection.$anchorCell.index();
+            const anchorCellRowIndex =
+              editor.prosemirrorState.selection.$anchorCell.index(-1);
+            const headCellColIndex =
+              editor.prosemirrorState.selection.$headCell.index();
+            const headCellRowIndex =
+              editor.prosemirrorState.selection.$headCell.index(-1);
+
+            const minColIndex = Math.min(anchorCellColIndex, headCellColIndex);
+            const maxColIndex = Math.max(anchorCellColIndex, headCellColIndex);
+            const minRowIndex = Math.min(anchorCellRowIndex, headCellRowIndex);
+            const maxRowIndex = Math.max(anchorCellRowIndex, headCellRowIndex);
+
+            if (
+              minColIndex !== 0 ||
+              maxColIndex !== cols - 1 ||
+              minRowIndex !== 0 ||
+              maxRowIndex !== rows - 1
+            ) {
+              return false;
+            }
+
+            const selectionBlock =
+              editor.getPrevBlock(block) || editor.getNextBlock(block);
+            if (selectionBlock) {
+              editor.setTextCursorPosition(block);
+            }
+
+            editor.removeBlocks([block]);
```

Comment:
Would prefer for this to run in an `editor.transact` so that it is only a single undo step.